rust/datafusion/src/datasource/parquet.rs

```diff
@@ -262,20 +263,55 @@ impl ParquetFile {
                                 is_nullable,
                             )?
                         }
-                        ColumnReader::Int32ColumnReader(ref mut r) => {
-                            ArrowReader::<Int32Type>::read(
+                        ColumnReader::Int32ColumnReader(ref mut r) => match dt {
+                            DataType::Date32(DateUnit::Millisecond) => {
+                                ArrowReader::<Date32MillisecondType>::read(
+                                    r,
+                                    self.batch_size,
+                                    is_nullable,
+                                )?
+                            }
+                            DataType::Time32(TimeUnit::Millisecond) => {
+                                ArrowReader::<Time32MillisecondType>::read(
+                                    r,
+                                    self.batch_size,
+                                    is_nullable,
+                                )?
+                            }
+                            _ => ArrowReader::<Int32Type>::read(
                                 r,
                                 self.batch_size,
                                 is_nullable,
-                            )?
-                        }
-                        ColumnReader::Int64ColumnReader(ref mut r) => {
-                            ArrowReader::<Int64Type>::read(
+                            )?,
+                        },
+                        ColumnReader::Int64ColumnReader(ref mut r) => match dt {
+                            DataType::Time64(TimeUnit::Microsecond) => {
+                                ArrowReader::<Time64MicrosecondType>::read(
+                                    r,
+                                    self.batch_size,
+                                    is_nullable,
+                                )?
+                            }
+                            DataType::Timestamp(TimeUnit::Microsecond) => {
+                                ArrowReader::<TimestampMicrosecondType>::read(
+                                    r,
+                                    self.batch_size,
+                                    is_nullable,
+                                )?
+                            }
+                            DataType::Timestamp(TimeUnit::Millisecond) => {
```

Comment:
Please add TimestampNano. This will cover the successor of the deprecated `int96` type

Reply:
Do you know where I can get a parquet file with all the valid types so I can write comprehensive unit tests for this?

Reply:
maybe try https://github.com/apache/parquet-testing/blob/master/data/alltypes_plain.parquet

Reply:
this is the schema for "alltypes"

```
"id: Int32\n\
        bool_col: Boolean\n\
        tinyint_col: Int32\n\
        smallint_col: Int32\n\
        int_col: Int32\n\
        bigint_col: Int64\n\
        float_col: Float32\n\
        double_col: Float64\n\
        date_string_col: Utf8\n\
        string_col: Utf8\n\
        timestamp_col: Timestamp(Nanosecond)"
```

The only date/time is the INT96 one

Reply:
This seems like a good question for the mailing list